File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string r = "";
        int open = 0;
        for(char c : s)
        {
            if(c == '(' && open++ > 0) r += c;
            else if(c == ')' && open-- > 1) r+=c;
        }
        return r;
    }
};

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        r = ""
        ope = 0
        for i in s:
            if i == '(' and ope > 0:
                r += i
            elif i == ')' and ope > 1:
                r += i
            ope += 1 if i == '(' else -1
        return r